pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go

```diff
@@ -362,6 +362,7 @@ func machineDeployments(cluster *rancherv1.Cluster, capiCluster *capi.Cluster, d
 							},
 						},
 						InfrastructureRef: infraRef,
+						NodeDrainTimeout:  machinePool.DrainBeforeDeleteTimeout,
```

Comment:
We have this problem where every cluster will be reprovisioned when Rancher is upgraded. I think that if we "upgrade" from v2.6-head to this change, then we will not have this problem because `DrainBeforeDeleteTimeout` is a pointer.

Has this been tested with these changes?

Reply:
I haven't tested the upgrade case yet, but I will confirm.

Reply:
Confirmed this change does not trigger a redeployment